tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1426 @@
+      val authKey = wp.authKey.split(":")
+      new BasicHttpCredentials(authKey(0), authKey(1))
+    } else {
+      new BasicHttpCredentials(wp.authKey, wp.authKey)
+    }
+  }
+
+  def getAuthKey(wp: WskProps): String = {
+    val authKey = wp.authKey.split(":")
+    s"${authKey(0)}:${authKey(1)}"
+  }
+
+  def getParamsAnnos(parameters: Map[String, JsValue] = Map(),
+                     annotations: Map[String, JsValue] = Map(),
+                     parameterFile: Option[String] = None,
+                     annotationFile: Option[String] = None,
+                     feed: Option[String] = None,
+                     web: Option[String] = None): (JsArray, JsArray) = {
+    val params = parameterFile map { pf =>
+      convertStringIntoKeyValue(pf)
+    } getOrElse convertMapIntoKeyValue(parameters)
+    val annos = annotationFile map { af =>
+      convertStringIntoKeyValue(af, feed, web)
+    } getOrElse convertMapIntoKeyValue(annotations, feed, web)
+    (params, annos)
+  }
+
+  def convertStringIntoKeyValue(file: String, feed: Option[String] = None, web: Option[String] = None): JsArray = {
+    var paramsList = Vector[JsObject]()
+    val input = FileUtils.readFileToString(new File(file))
+    val in = input.parseJson.convertTo[Map[String, JsValue]]
+    convertMapIntoKeyValue(in, feed, web)
+  }
+
+  def convertMapIntoKeyValue(params: Map[String, JsValue],
+                             feed: Option[String] = None,
+                             web: Option[String] = None): JsArray = {
+    var paramsList = Vector[JsObject]()
+    params foreach { case (key, value) => paramsList :+= JsObject("key" -> key.toJson, "value" -> value.toJson) }
+    paramsList = feed map { f =>
+      paramsList :+ JsObject("key" -> "feed".toJson, "value" -> f.toJson)
+    } getOrElse paramsList
+    paramsList = web map { w =>
+      paramsList :+ JsObject("key" -> "web-export".toJson, "value" -> w.toJson)
+    } getOrElse paramsList
+    JsArray(paramsList)
+  }
+
+  def entityName(name: String)(implicit wp: WskProps) = {
+    val sep = "/"
+    if (name.startsWith(sep)) name.substring(name.indexOf(sep, name.indexOf(sep) + 1) + 1, name.length())
+    else name
+  }
+
+  def fullEntityName(name: String)(implicit wp: WskProps) = {
+    val sep = "/"
+    if (name.startsWith(sep)) name
+    else s"/${wp.namespace}/$name"
+  }
+
+  def convertIntoComponents(comps: String)(implicit wp: WskProps): JsArray = {
+    var paramsList = Vector[JsString]()
+    comps.split(",") foreach {
+      case (value) =>
+        val fullName = fullEntityName(value)
+        paramsList :+= JsString(fullName)
+    }
+    JsArray(paramsList)
+  }
+
+  def getRespData(resp: HttpResponse): String = {
+    val timeout = 5.seconds
+    Try {
+      resp.entity.toStrict(timeout).map { _.data }.map(_.utf8String).futureValue
+    } getOrElse {
+      ""
+    }
+  }
+
+  def getNamespaceActionName(name: String)(implicit wp: WskProps): (String, String) = {
```

Comment:
Is this equivalent?
```
def getNamespaceEntityName(quailfiedName: String)(implicit wp: WskProps): (Option[String], Option[String]) = {
	quailfiedName.split("/") match {
		case Array(empty, namespace, entityName) => (Some(namespace), Some(entityName))
		case Array(empty, namespace) => (Some(namespace), None)
		case Array(entityName) => (Some(wp.namespace), Some(entityName))
		case _ => (None, None)
	}
}
```

Reply:
Maybe throw an error on `_` case. What do you think?

Reply:
Let me see how to rewrite it. 
There is no need to fo case _ => (None, None), because the worst case is to return wp.namespace and name.